### PR TITLE
Jack toolchain and lambdas

### PR DESCRIPTION
**Pull request by florina-muntenescu**

### CODE REVIEW

File: todoapp/app/build.gradle [hunk 1]
```diff
@@ -12,11 +12,20 @@ android {
...
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
     }
 
     buildTypes {
         debug {
-            minifyEnabled true
+            minifyEnabled false
```
Comment (JoseAlcerreca) [inline todoapp/app/build.gradle]: on purpose?

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/TasksRepository.java [hunk 2]
```diff
@@ -118,12 +118,7 @@ public static void destroyInstance() {
             // Query the local storage if available. If not, query the network.
             Observable<List<Task>> localTasks = getAndCacheLocalTasks();
             return Observable.concat(localTasks, remoteTasks)
-                    .filter(new Func1<List<Task>, Boolean>() {
-                        @Override
-                        public Boolean call(List<Task> tasks) {
-                            return !tasks.isEmpty();
-                        }
-                    }).first();
+                    .filter(tasks -> !tasks.isEmpty()).first();
```
Comment (JoseAlcerreca) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/TasksRepository.java]: nit: move .first to new line.

File: todoapp/app/build.gradle (hunk 1 above)
Comment (florina-muntenescu) [inline todoapp/app/build.gradle]: Yes. With `minifyEnabled true`, when building, you get the error 
`Minifying the variant used for tests is not supported when using Jack.`

Comment (JoseAlcerreca) [inline todoapp/app/build.gradle]: Uhmm in that case comment and at least minify the release build type. http://stackoverflow.com/questions/37314760/android-how-to-perform-obfuscation-with-the-jack-compiler

It's important to compare APK size and method count.

**Pull request closed without merging**